### ISSUE TITLE
False positive on Style/CommentedKeyword inside class_eval

### ISSUE THREAD

**Issue by woodruffw** [bbatsov/rubocop#4945]
I'm getting a false positive from the `Style/CommentedKeyword` cop in one of my codebases. It looks like RuboCop gets confused by the combination of a `class_eval` and a `%[]` string. Details below.

-------

## Expected behavior

No Style/CommentedKeyword warning.

## Actual behavior

```
lib/kbsecret/record/abstract.rb:55:17: C: Style/CommentedKeyword: Do not place comments on the same line as the def keyword.
            def #{field}
                ^^^^^^^^
```

## Steps to reproduce the problem

This is the most minimal example I could come up with:

```ruby
class_eval %[
  def #{field}
    "foo"
  end
]
```

The actual offending code in my own codebase is here: https://github.com/kbsecret/kbsecret/blob/master/lib/kbsecret/record/abstract.rb#L53

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.2 x86_64-linux-gnu)
```

**Comment by pocke** [comment]
duplicated. https://github.com/bbatsov/rubocop/issues/4886
And this problem will be fixed by https://github.com/bbatsov/rubocop/pull/4944
